### PR TITLE
1504 build locally without socks

### PR DESCRIPTION
**Pull request by micahchiang**
## Description
As a new engineer waiting for access to the SOCKS Proxy, I would like to build locally so I can work in the mean time. Currently, running `yarn watch` fails because there is no cached content available and querying drupal requires a SOCKS proxy connection. 

## Testing done
Tested locally that `yarn watch` builds as expected.

## Screenshots
Build failure:
Screen Shot 2019-09-05 at 3 38 54 PM

## Acceptance criteria
- [x] `yarn watch` builds properly when no `.cache` directory exists
- [x] vets-website is accessible locally.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by micahchiang** [comment]
@cvalarida - tagging you since you're familiar with the issue. Let me know if anything else is needed. Thanks again for the help!

File: src/site/stages/build/drupal/metalsmith-drupal.js [hunk 1]
```diff
@@ -26,13 +26,18 @@ const PULL_DRUPAL_BUILD_ARG = 'pull-drupal';
 
 // We need to pull the Drupal content if we have --pull-drupal OR if
 // the content is not available in the cache.
+
```
Comment (cvalarida) [inline src/site/stages/build/drupal/metalsmith-drupal.js]:
```suggestion
```

File: src/site/stages/build/drupal/load-saved-flags.js [hunk 2]
```diff
@@ -51,12 +55,14 @@ function useFlags(rawFlags) {
  * This is for use with the drupal-aws-cache script. The normal build
  * and preview build scripts do this in setUpFeatureFlags().
  */
-function loadFeatureFlags(cacheDirectory) {
+function loadFeatureFlags(cacheDirectory, buildType) {
   const featureFlagFile = path.join(cacheDirectory, 'feature-flags.json');
-  const rawFlags = fs.readJsonSync(featureFlagFile);
+  const rawFlags = fs.existsSync(featureFlagFile)
+    ? fs.readJsonSync(featureFlagFile)
+    : {};
```
Comment (cvalarida) [inline src/site/stages/build/drupal/load-saved-flags.js]: I'm wondering if this is necessary. The only time `loadFeatureFlags` is called is in the `drupal-aws-cache.js` script. That script is called to either load from or save the cache to the S3 bucket. If we're trying to load the cache, we'll need the "real" feature flags in order to construct the right query to get the cache key. If we're trying to save the cache to S3, we need the right flags to save the cache with the right key. :thinking: 

With that in mind, I think it's okay to fail the script if the feature flags file isn't found.

It's probably worth adding a comment to the function saying as much.

File: src/site/stages/build/drupal/metalsmith-drupal.js [hunk 3]
```diff
@@ -26,13 +26,18 @@ const PULL_DRUPAL_BUILD_ARG = 'pull-drupal';
...
   const drupalCache = path.join(
     buildOptions.cacheDirectory,
     DRUPAL_CACHE_FILENAME,
   );
   const isDrupalAvailableInCache = fs.existsSync(drupalCache);
-  return buildOptions[PULL_DRUPAL_BUILD_ARG] || !isDrupalAvailableInCache;
+  return (
+    buildOptions[PULL_DRUPAL_BUILD_ARG] ||
+    (!isDrupalAvailableInCache &&
+      buildOptions.buildtype !== ENVIRONMENTS.LOCALHOST)
```
Comment (cvalarida) [inline src/site/stages/build/drupal/metalsmith-drupal.js]:
```suggestion
      buildOptions.buildtype !== ENVIRONMENTS.LOCALHOST) // Don't require a cache to build locally
```

File: src/site/stages/build/drupal/load-saved-flags.js [hunk 4]
```diff
@@ -27,7 +28,10 @@ function useFlags(rawFlags) {
         'inspect',
         'Symbol(Symbol.iterator)',
       ];
-      if (!ignoreList.includes(prop.toString())) {
+      if (
+        !ignoreList.includes(prop.toString()) &&
+        buildType !== ENVIRONMENTS.LOCALHOST // Don't require a cache to build locally.
```
Comment (cvalarida) [inline src/site/stages/build/drupal/load-saved-flags.js]: This one's a little different. My suggestion to add this comment was in the `shouldPullDrupal` function.
```suggestion
        buildType !== ENVIRONMENTS.LOCALHOST // Don't fail a localhost build for missing query flags
```

**Review by cvalarida — APPROVED** [review]
Just a couple in-code comments to clarify some boolean logic and I think we're golden!

**Comment by cvalarida** [comment]
cc @ncksllvn to make sure we're not missing anything.

File: src/site/stages/build/drupal/load-saved-flags.js (hunk 4 above)
Comment (micahchiang) [reply src/site/stages/build/drupal/load-saved-flags.js]: Doh. Sorry I'll update.

**Pull request merged**